packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -222,67 +221,90 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
         _token: vscode.CancellationToken
     ): Promise<void> {
         const droppedItems = dataTransfer.get("application/vnd.code.tree.zowe.ds.explorer");
-        if (!droppedItems) {
-            return;
-        }
+        if (!droppedItems) return;
 
         let target = targetNode;
-        if (target) {
-            for (const item of droppedItems.value) {
-                const node = this.draggedNodes[item.uri.path];
-
-                if (SharedContext.isPds(target) || SharedContext.isDsMember(target)) {
-                    if (SharedContext.isPds(node) || SharedContext.isDs(node)) {
-                        Gui.errorMessage(
-                            vscode.l10n.t("Cannot drop a sequential dataset or a partitioned dataset into another partitioned dataset.")
-                        );
-                        return;
-                    }
-                }
+        if (!target) return;
+
+        // check each dropped node for same-object, member collision, and structure issues
+        for (const item of droppedItems.value) {
+            const node = this.draggedNodes[item.uri.path];
+            if (!node) continue;
+            const srcDsn = SharedUtils.getNodeProperty(node, "label");
+
+            // 1. Same-object check
+            if (
+                node.getProfile &&
+                target.getProfile &&
+                await SharedUtils.isSamePhysicalDataset(
+                    node.getProfile(),
+                    target.getProfile(),
+                    srcDsn
+                )
+            ) {
+                Gui.errorMessage(vscode.l10n.t(SharedUtils.ERROR_SAME_OBJECT_DROP));
```

Comment:
Since this string was changed, it should trigger an update in the l10n files. Please run `pnpm --filter vscode-extension-for-zowe vscode:prepublish`